src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -106,18 +106,7 @@ void ICompilation.Compile(string outputFile)
             // TODO: Hacky static field
 
             string systemModuleName = ((IAssemblyDesc)NodeFactory.TypeSystemContext.SystemModule).GetName().Name;
-
-            // TODO: CompilationUnitPrefix is used even before this point!!!
-            // TODO: just something to get Runtime.Base compiled
-            if (systemModuleName != "System.Private.CoreLib")
-            {
-                NodeFactory.CompilationUnitPrefix = systemModuleName.Replace(".", "_");
-            }
-            else
-            {
-                NodeFactory.CompilationUnitPrefix = NameMangler.SanitizeName(Path.GetFileNameWithoutExtension(outputFile));
-            }
-
+            
```

Comment:
Seems like we can now delete everything but the `CompileInternal` call from here.

(These are the places where I wish C# supported abstract explicit interface method implementations - we could get rid of `CompileInternal` that way...)